test/jdk/java/nio/channels/FileChannel/ChannelInputStreamTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 8361495
+ * @summary Test for AsynchronousCloseException from uninterruptible FileChannel
+ * @run junit ChannelInputStreamTest
+ */
+
+import java.io.Closeable;
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.channels.AsynchronousCloseException;
+import java.nio.channels.ClosedChannelException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import org.junit.jupiter.api.Test;
+
+public class ChannelInputStreamTest {
+    private static Thread startCloseThread(ConcurrentLinkedQueue<Object> q) {
+        return Thread.ofPlatform().start(() -> {
+            do {
+                Object obj;
+                if ((obj = q.poll()) != null) {
+                    if (obj instanceof Closeable c) {
+                        try {
+                            c.close();
+                        } catch (IOException ignored) {
+                        }
+                    } else {
+                        break;
+                    }
+                }
+            } while (true);
+        });
+    }
+
+    @Test
+    public void available() throws IOException {
+        var close = new ConcurrentLinkedQueue<Object>();
+        Thread closeThread = startCloseThread(close);
```

Comment:
I think clear to put a try-finally after this, where the finally does the closeThread.join.